script/tool/lib/src/firebase_test_lab_command.dart

```diff
@@ -105,183 +101,165 @@ class FirebaseTestLabCommand extends PluginCommand {
+  /// the build target.
+  ///
+  /// Returns true if the command succeeds.
+  Future<bool> _runGradle(
+    Directory directory,
+    String target, {
+    File? testFile,
+  }) async {
+    final String experiment = getStringArg(kEnableExperiment);
+    final String? extraOptions = experiment.isNotEmpty
+        ? Uri.encodeComponent('--enable-experiment=$experiment')
+        : null;
 
-          if (exitCode != 0) {
-            failingPackages.add(packageName);
-            continue;
-          }
-          final String buildId = getStringArg('build-id');
-          final String testRunId = getStringArg('test-run-id');
-          final String resultsDir =
-              'plugins_android_test/$packageName/$buildId/$testRunId/${resultsCounter++}/';
-          final List<String> args = <String>[
-            'firebase',
-            'test',
-            'android',
-            'run',
-            '--type',
-            'instrumentation',
-            '--app',
-            'build/app/outputs/apk/debug/app-debug.apk',
-            '--test',
-            'build/app/outputs/apk/androidTest/debug/app-debug-androidTest.apk',
-            '--timeout',
-            '5m',
-            '--results-bucket=${getStringArg('results-bucket')}',
-            '--results-dir=$resultsDir',
-          ];
-          for (final String device in getStringListArg('device')) {
-            args.addAll(<String>['--device', device]);
-          }
-          exitCode = await processRunner.runAndStream('gcloud', args,
-              workingDir: exampleDirectory);
+    final int exitCode = await processRunner.runAndStream(
+        p.join(directory.path, _gradleWrapper),
+        <String>[
+          target,
+          '-Pverbose=true',
+          if (testFile != null) '-Ptarget=${testFile.path}',
+          if (extraOptions != null) '-Pextra-front-end-options=$extraOptions',
+          if (extraOptions != null)
+            '-Pextra-gen-snapshot-options=$extraOptions',
+        ],
+        workingDir: directory);
 
-          if (exitCode != 0) {
-            failingPackages.add(packageName);
-            continue;
-          }
-        }
-      }
+    if (exitCode != 0) {
+      return false;
     }
+    return true;
+  }
 
-    _print('\n\n');
-    if (failingPackages.isNotEmpty) {
-      _print(
-          'The instrumentation tests for the following packages are failing (see above for'
-          'details):');
-      for (final String package in failingPackages) {
-        _print(' * $package');
-      }
-    }
-    if (missingFlutterBuild.isNotEmpty) {
-      _print('Run "pub global run flutter_plugin_tools build-examples --apk" on'
-          'the following packages before executing tests again:');
-      for (final String package in missingFlutterBuild) {
-        _print(' * $package');
-      }
-    }
+  /// Finds and returns all integration test files for [package].
+  List<File> _findIntegrationTestFiles(Directory package) {
```

Comment:
nit: This function would be more efficient as a generator.

Reply:
Done. (In practice this is usually only returning one thing, but I only didn't use a generator because I often forget that's a thing in Dart.)